src/include/evict.h

```diff
@@ -0,0 +1,174 @@
+    uint64_t cache_stuck_timeout_ms; /* Maximum time the cache can be stuck for in diagnostic mode
+                                        before timing out */
+
+    /*
+     * Eviction thread tuning information.
+     */
+    uint32_t evict_tune_datapts_needed;          /* Data needed to tune */
+    struct timespec evict_tune_last_action_time; /* Time of last action */
+    struct timespec evict_tune_last_time;        /* Time of last check */
+    uint32_t evict_tune_num_points;              /* Number of values tried */
+    uint64_t evict_tune_progress_last;           /* Progress counter */
+    uint64_t evict_tune_progress_rate_max;       /* Max progress rate */
+    bool evict_tune_stable;                      /* Are we stable? */
+    uint32_t evict_tune_workers_best;            /* Best performing value */
+
+    /*
+     * Pass interrupt counter.
+     */
+    wt_shared volatile uint32_t pass_intr; /* Interrupt eviction pass. */
+
+    /*
+     * LRU eviction list information.
+     */
+    WT_SPINLOCK evict_pass_lock;   /* Eviction pass lock */
+    WT_SESSION_IMPL *walk_session; /* Eviction pass session */
+    WT_DATA_HANDLE *walk_tree;     /* LRU walk current tree */
+
+    WT_SPINLOCK evict_queue_lock; /* Eviction current queue lock */
+    WT_EVICT_QUEUE evict_queues[WT_EVICT_QUEUE_MAX];
+    WT_EVICT_QUEUE *evict_current_queue; /* LRU current queue in use */
+    WT_EVICT_QUEUE *evict_fill_queue;    /* LRU next queue to fill.
+                                            This is usually the same as the
+                                            "other" queue but under heavy
+                                            load the eviction server will
+                                            start filling the current queue
+                                            before it switches. */
+    WT_EVICT_QUEUE *evict_other_queue;   /* LRU queue not in use */
+    WT_EVICT_QUEUE *evict_urgent_queue;  /* LRU urgent queue */
+    uint32_t evict_slots;                /* LRU list eviction slots */
+
+#define WT_EVICT_PRESSURE_THRESHOLD 0.95
+#define WT_EVICT_SCORE_BUMP 10
+#define WT_EVICT_SCORE_CUTOFF 10
+#define WT_EVICT_SCORE_MAX 100
+    /*
+     * Score of how aggressive eviction should be about selecting eviction candidates. If eviction
+     * is struggling to make progress, this score rises (up to a maximum of WT_EVICT_SCORE_MAX), at
+     * which point the cache is "stuck" and transactions will be rolled back.
+     */
+    wt_shared uint32_t evict_aggressive_score;
+
+    /*
+     * Score of how often LRU queues are empty on refill. This score varies between 0 (if the queue
+     * hasn't been empty for a long time) and 100 (if the queue has been empty the last 10 times we
+     * filled up.
+     */
+    uint32_t evict_empty_score;
+
+/*
+ * Flags.
+ */
+/* AUTOMATIC FLAG VALUE GENERATION START 0 */
+#define WT_EVICT_CACHE_CLEAN 0x001u        /* Evict clean pages */
```

Comment:
nit: Do we need `_CACHE_` in these macros? `WT_EVICT_CLEAN`/`WT_EVICT_DIRTY_HARD`/`WT_EVICT_URGENT` (for a few examples) seem readable to me.

Reply:
I’m open to removing it and can make the changes if you’d like. CACHE helps convey that we are evaluating things from the cache, but that’s somewhat implied anyway.

Let me know if you'd like me to remove it.

Reply:
Not enough to push for it. If you're happy with it let's keep it 👍 . Resolving